pkg/client/cache/delta_fifo.go

```diff
@@ -522,6 +522,23 @@ func (f *DeltaFIFO) syncKey(key string) error {
 		return nil
 	}
 
+	// If we are doing Resync() and there is already an event queued for that object,
```

Comment:
That's probably wrong. If it changes a value "sync" is probably wrong. Scary to change, though.